compiler/rustc_expand/src/proc_macro.rs

```diff
@@ -124,36 +132,38 @@ impl MultiItemModifier for DeriveProcMacro {
         // altogether. See #73345.
         crate::base::ann_pretty_printing_compatibility_hack(&item, &ecx.sess);
         let input = item.to_tokens();
-        let stream = {
-            let _timer =
-                ecx.sess.prof.generic_activity_with_arg_recorder("expand_proc_macro", |recorder| {
-                    recorder.record_arg_with_span(
-                        ecx.sess.source_map(),
-                        ecx.expansion_descr(),
-                        span,
-                    );
-                });
-            let proc_macro_backtrace = ecx.ecfg.proc_macro_backtrace;
-            let strategy = exec_strategy(ecx);
-            let server = proc_macro_server::Rustc::new(ecx);
-            match self.client.run(&strategy, server, input, proc_macro_backtrace) {
-                Ok(stream) => stream,
-                Err(e) => {
-                    ecx.dcx().emit_err({
-                        errors::ProcMacroDerivePanicked {
-                            span,
-                            message: e.as_str().map(|message| {
-                                errors::ProcMacroDerivePanickedHelp { message: message.into() }
-                            }),
-                        }
-                    });
-                    return ExpandResult::Ready(vec![]);
+        let res = ty::tls::with(|tcx| {
+            // FIXME(pr-time): without flattened some (weird) tests fail, but no idea if it's correct/enough
```

Comment:
In theory flattening should always be correct (proc macro APIs will perform flattening lazily when necessary), but in practice there may be issues with badly written proc macros that pretty-print token streams and then parse the result as a text.

Reply:
Mhhm, makes sense. What would you suggest for now? I think caching needs the flattening, so should I try to not flatten when caching is disabled, that is, use the same behavior as before this PR?